browse/util/id_patterns.py

```diff
@@ -0,0 +1,323 @@
+r"""Patterns and functions to detect arXiv ids and Urls in text.
```

Comment:
DRY and `arxiv.identifier.parse_arxiv_id` - perhaps some of these functions should also be folded into [`arxiv.identifier` in arxiv-base](https://github.com/cul-it/arxiv-base/blob/master/arxiv/identifier/__init__.py)?